src/Directory.Build.props

```diff
@@ -5,4 +5,9 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
+  <ItemGroup>
```

Comment:
why are these required?

Reply:
This brings two significant benefits:

- It reduces the number of times you have to add nullability suppressions (`!` operator) because it clues the compiler in to e.g. the fact that `string.IsNullOrEmpty` is a null check.
- It catches places where the BCL is documented to sometimes return null which would cause Shouldly to throw NullReferenceException.

Reply:
ok cool. but given nuget ui doesnt yet recognize solutionlevel packages, do u think it would be better (for the short term) to make them project level to make it easier to update them?

Reply:
I found some superfluous existing uses of `!` that can be removed now that the BCL is annotated. I'll push that soon.

Reply:
Two commits pushed actually.

Reply:
> ok cool. but given nuget ui doesnt yet recognize solutionlevel packages, do u think it would be better (for the short term) to make them project level to make it easier to update them?

Forgot to refresh, just saw this. I am used to updating those by copying and pasting the PackageReference tag from the NuGet page, but I do get what you mean. Sounds good. I'll push a commit for this next.

(There's something slightly different than Directory.Build.props around the corner, [Directory.Packages.props](https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions), but that's not shipped yet.)

Reply:
Done.